ecl/hqlcpp/hqlcppds.cpp

```diff
@@ -2455,6 +2455,30 @@ void HqlCppTranslator::doBuildDataset(BuildCtx & ctx, IHqlExpression * expr, CHq
     }
     else
     {
+        IHqlExpression * record = expr->queryRecord();
+        IAtom * serializeForm = internalAtom; // The format of serialized expressions in memory must match the internal serialization format
+        OwnedHqlExpr serializedRecord = getSerializedForm(record, serializeForm);
+        if (format == FormatNatural)
+        {
+            EvalContext * instance = queryEvalContext(ctx);
+            if (instance)
```

Comment:
essentially this code was moved to happen earlier.  Only change is line line and the next which were added to avoid a slight decrease in code quality for example issue16893.eclxml.